src/pages/billing/List/CreateBillingProjectControl.ts

```diff
@@ -4,33 +4,39 @@ import { ButtonOutline } from 'src/components/common'
 import { icon } from 'src/components/icons'
 import { MenuButton } from 'src/components/MenuButton'
 import { MenuTrigger } from 'src/components/PopupTrigger'
+import { CloudProvider } from 'src/libs/workspace-utils'
 import { cloudProviders } from 'src/pages/workspaces/workspace/analysis/utils/runtime-utils'
 
 
-export const CreateBillingProjectControl = ({ isAzurePreviewUser, showCreateProjectModal }) => {
-  const createButton = (onClickCallback, type) => {
+interface CreateBillingProjectControlProps {
+  isAzurePreviewUser: boolean
+  showCreateProjectModal: (type: { label: CloudProvider }) => void
```

Comment:
I am not crazy about this type definition, but this is how `cloudProviders.gcp` and `cloudProviders.azure` are defined.